### PR TITLE
Added Scons Option for the Mold Linker

### PR DESCRIPTION
**Pull request by cyanff**
As proposed in https://github.com/godotengine/godot-proposals/issues/4531
This should lead to slightly faster link times.

Note that while mold is fully supported in Clang, on GCC it's only supported on the latest version (12.1.0).

### CODE REVIEW

**Comment by Chaosus** [comment]
You need to squash commits as required by our pipeline (https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html)

**Comment by thisisphan** [comment]
Squashed!

File: platform/linuxbsd/detect.py [hunk 1]
```diff
@@ -89,7 +90,7 @@ def get_flags():
     return []
 
 
-def configure(env):
+def configure(env, cc_version_int):
```
Comment (bruvzg) [inline platform/linuxbsd/detect.py:93]: Since you have added a new argument to the `configure`, the same change should be done for all other platforms.

**Review by bruvzg — COMMENTED** [review]
Note: `mold` do support macOS as well, but it's in the alpha stage, and currently fails to link Godot, so there's no reason for adding it to the macOS build config for now.

File: platform/linuxbsd/detect.py (hunk 1 above)
Comment (akien-mga) [reply platform/linuxbsd/detect.py:93]: I'm not fond of adding this to the prototype. You can just get this info from methods.py where needed.

Comment (thisisphan) [reply platform/linuxbsd/detect.py]: > I'm not fond of adding this to the prototype. You can just get this info from methods.py where needed.

Thanks for pointing this out, I've pushed a fix to this.

**Comment by Calinou** [comment]
>  Is mold currently supported for any other platforms other than linuxbsd?

Mold currently only supports UNIX-based platforms, so it doesn't support Windows.

**Comment by akien-mga** [comment]
Sorry I had forgotten about this PR when I just decided to do my own implementation for this today: #61433.

I think my solution is a bit better as it also enables selecting GNU gold if wanted, as well as using LLD on Linux (which works, contrarily to what our error message was saying).

Thanks for the contribution nevertheless!

**Pull request closed without merging**